File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        maxave = ave = float("-inf")
        sum = 0

        l = r = 0
        while r < len(nums):
            while r - l < k:
                sum += nums[r]
                r += 1
            ave = sum / k
            ave = round(ave, 5)
            
            maxave = max(maxave, ave)
            sum -= nums[l]
            l += 1
        return maxave

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        // we should use double insted of float since we want precision
        double maxave = -INFINITY; // using the cmath library, for double type we use can use INFINITY but for very big numbers useHUGE_VAL
        double ave = -INFINITY;
        int sum = 0;
        int l = 0, r = 0;
        while (r < nums.size())
        {
            while (r - l < k)
            {
                sum += nums[r];
                r++;
            }
            ave = (double) sum / k;
            // cout << fixed << setprecision(5);
            ave = round(ave * 100000.0) / 100000.0;
            maxave = max(maxave, ave);
            sum -= nums[l];
            l++;
        }
        return maxave;
    }
};